### PR TITLE
The Mass Driver Update

### PR DESCRIPTION
**Pull request by DeityLink**
massdriverupdate
- Mass Drivers and their buttons are now buildable and deconstructable.
- Mass Drivers can now be emagged, quintupling the velocity at which they throw things
- Completely revamped the "Pod Doors and Mass Drivers Control" computer, it can now store multiple ID_Tags, and even fire all of their respective drivers simultaneously!
- Made a Deathsquad version of this computer, that can operate the Mass Drivers in centcomm and send the mechs to various ZLevels.
- Buttons can trigger a Mass Driver at a maximal range of 7, Computers cover the entire Z-Level
- Moved the piano back inside the bar, as was often requested
- Added a few Entertainment Monitors in a handful of fitting place throughout the station
- Made a new sprite for the Tcomms Server computer (was previously using the sprite of the PDA Messages monitoring computer)

Might find a way to build pod doors in the futur, now that we have an easy way to toggle them.

### CODE REVIEW

**Comment by clusterfack** [comment]
Dem lines changed

**Comment by DeityLink** [comment]
I forgot to clean the maps, >_>

**Comment by clusterfack** [comment]
This isn't a nitpick about your pull but we need to really to offload mounted frames to a secondary menu just for them alone soon because they are becoming quite a big list.

**Comment by DeityLink** [comment]
aye, we'll have to find something ergonomic.

File: code/game/machinery/mass_driver.dm [hunk 1]
```diff
@@ -9,34 +8,198 @@
...
+				playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+				if(do_after(user, 20) && (C.amount >= 3))
+					C.use(3)
+					user << "<span class='notice'>You've added cables to \the [src].</span>"
+					build++
+					update_icon()
+		if(3) // Wired
+			if(istype(W, /obj/item/weapon/wirecutters))
+				user << "You begin to remove the wiring from \the [src]."
+				if(do_after(user, 10))
```
Comment (ComicIronic) [inline code/game/machinery/mass_driver.dm]: Busy check

File: code/game/machinery/mass_driver.dm [hunk 2]
```diff
@@ -9,34 +8,198 @@
...
+					playsound(get_turf(src), 'sound/items/Wirecutter.ogg', 50, 1)
+					user << "<span class='notice'>You've removed the cables from \the [src].</span>"
+					build--
+					update_icon()
+				return 1
+			if(istype(W, /obj/item/stack/rods))
+				var/obj/item/stack/rods/R=W
+				user << "You begin to complete \the [src]..."
+				playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+				if(do_after(user, 20) && (R.amount >= 2))
```
Comment (ComicIronic) [inline code/game/machinery/mass_driver.dm]: Busy check

**Comment by clusterfack** [comment]
The airlock frames actually already have a secondary menu, but I do bemoan the lack of a back button.

File: code/game/machinery/mass_driver.dm [hunk 3]
```diff
@@ -9,34 +8,198 @@
...
+/obj/machinery/mass_driver_frame/attackby(var/obj/item/W as obj, var/mob/user as mob)
+	switch(build)
+		if(0) // Loose frame
+			if(istype(W, /obj/item/weapon/weldingtool))
+				var/obj/item/weapon/weldingtool/WT = W
+				if(!WT.remove_fuel(0, user))
+					user << "The welding tool must be on to complete this task."
+					return 1
+				playsound(get_turf(src), 'sound/items/Welder.ogg', 50, 1)
+				usr << "You begin to cut the frame apart..."
```
Comment (clusterfack) [inline code/game/machinery/mass_driver.dm]: user

**Comment by Exxion** [comment]
You thinking what I'm thinking?


image

**Comment by Manaony** [comment]
Instead of making another use for the Emag why not just having them hackable?

**Comment by DeityLink** [comment]
That'd mean adding wires and some wire datums. Maybe later.

**Comment by DeityLink** [comment]
* Added coat racks to detective offices on all three main stations.
* Added piano stools.
* The Control Computers' timers can now be set to loop individually. The ride never ends!
* if you increase a Mass Driver's power, it'll now also consume more power at each launch.

**Comment by ComicIronic** [comment]
Muh atomic PRs

**Comment by DeityLink** [comment]
It's the "Mass Driver Update" with in small letters in the corner "and also all the small little thing I wanted to do if I ever had the chance to make a map update which I usually don't since there's always a pending Map-related PR and also before we enter the Feature Freeze"... >_>

**Comment by d3athrow** [comment]
Closed by be4f5e76f25a75ed09cff693d9c20a7a5e0a8f6f

**Pull request closed without merging**